src/OrleansAzureUtils/Providers/Streams/AzureQueue/AzureQueueBatchContainer.cs

```diff
@@ -58,6 +62,70 @@ private AzureQueueBatchContainer(Guid streamGuid, String streamNamespace, List<o
             this.requestContext = requestContext;
         }
 
+        /// <summary>
+        /// Creates a deep copy of an object
+        /// </summary>
+        /// <param name="original">The object to create a copy of</param>
+        /// <returns>The copy.</returns>
+        public static object DeepCopy(object original)
+        {
+            var source = original as AzureQueueBatchContainer;
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(original));
+            }
+
+            var token = source.sequenceToken == null ? null : new EventSequenceToken(source.sequenceToken.SequenceNumber, source.sequenceToken.EventIndex);
+            var events = source.events?.Select(SerializationManager.DeepCopy).ToList();
+            var context = source.requestContext?.ToDictionary(kv => kv.Key, kv => SerializationManager.DeepCopy(kv.Value));
+            return new AzureQueueBatchContainer(source.StreamGuid, source.StreamNamespace, events, context, token);
+        }
+
+        /// <summary>
+        /// Serializes the container to the binary stream.
+        /// </summary>
+        /// <param name="untypedInput">The object to serialize</param>
+        /// <param name="writer">The stream to write to</param>
+        /// <param name="expected">The expected type</param>
+        public static void Serialize(object untypedInput, BinaryTokenStreamWriter writer, Type expected)
+        {
+            var typed = untypedInput as AzureQueueBatchContainer;
+            if (typed == null)
+            {
+                throw new SerializationException();
+            }
+
+            writer.Write(typed.StreamGuid);
+            writer.Write(typed.StreamNamespace);
+            SerializationManager.Serialize(typed.sequenceToken, writer);
```

Comment:
These should be `SerializeInner` - similarly for the Deserialize calls. Check out the pattern used in auto-generated serializers, particularly for DeepCopy with its call to `SerializationContext.Current.RecordObject(original, result);` immediately after the result is created and for Deserializer with its call to `Serialization.`DeserializationContext.`Current.`RecordObject(result);` immediately after the constructor.